evap/static/ts/src/sortable-form.ts

```diff
@@ -0,0 +1,102 @@
+    formTemplate: string | null = null;
+}
+
+interface JQuery {
+    formset: (arg: FormsetOptions) => void;
+}
+
+function makeFormSortable(
+    tableId: string,
+    prefix: string,
+    rowChanged: (arg: HTMLTableRowElement) => boolean,
+    rowAdded: (arg: HTMLTableRowElement) => void,
+    removeAsButton: boolean,
+    usesTemplate: boolean,
+) {
+    function applyOrdering() {
+        document.querySelectorAll("tr").forEach((tableRow, i) => {
+            if (rowChanged(tableRow)) {
+                (tableRow.querySelectorAll("input[id$=-order]") as NodeListOf<HTMLInputElement>).forEach(input => {
+                    input.value = i.toString();
+                });
+            } else {
+                // if the row is empty (has no text in the input fields) set the order to -1 (default),
+                // so that the one extra row doesn't change its initial value
+                (tableRow.querySelectorAll("input[id$=-order]") as NodeListOf<HTMLInputElement>).forEach(input => {
+                    input.value = "-1";
+                });
+            }
+        });
+    }
+
+    // This is the only remaining jQuery usage, since formset requires jQuery
+    $(`#${tableId} tbody tr`).formset({
+        prefix: prefix,
+        deleteCssClass: removeAsButton ? "btn btn-danger btn-sm" : "delete-row",
+        deleteText: removeAsButton ? '<span class="fas fa-trash"></span>' : window.gettext("Delete"),
+        addText: window.gettext("add another"),
+        added: function (rowJQuery: JQuery<HTMLTableRowElement>) {
+            const row = rowJQuery.get()[0];
+            (row.querySelectorAll("input[id$=-order]") as NodeListOf<HTMLInputElement>).forEach(input => {
+                input.value = row.parentElement?.childElementCount.toString() ?? "";
+            });
+
+            // We have to empty the formset, otherwise sometimes old contents from
+            // invalid forms are copied (#644).
+            // Checkboxes with 'data-keep' need to stay checked.
+            row.querySelectorAll("input[type=checkbox]:not([data-keep]),input[type=radio]").forEach(el => {
+                el.removeAttribute("checked");
+            });
+
+            (row.querySelectorAll("input[type=text],input[type=textarea]") as NodeListOf<HTMLInputElement>).forEach(
+                input => {
+                    input.value = "";
+                },
+            );
+
+            row.querySelectorAll("select").forEach(el => {
+                el.querySelectorAll("option[selected]").forEach(option => {
+                    option.removeAttribute("selected");
+                });
+                el.querySelector("option")?.setAttribute("selected", "selected");
+            });
+
+            //Check the first item in every button group
+            row.querySelectorAll(".btn-group").forEach(group => {
+                group.querySelector("input")?.setAttribute("checked", "checked");
```

Comment:
Could it be that we need to do this instead?
```suggestion
                group.querySelector("input")?.checked = true;
```
Ref: https://stackoverflow.com/a/27906003